pkg/envoy/server.go

```diff
@@ -453,25 +482,41 @@ func getDirectionNetworkPolicy(l4Policy policy.L4PolicyMap, policyEnforced bool,
 			Rules:    make([]*cilium.PortNetworkPolicyRule, 0, len(l4.L7RulesPerEp)),
 		}
 
+		var allowAll bool
 		for sel, l7 := range l4.L7RulesPerEp {
-			rule := getPortNetworkPolicyRule(sel, l4.L7Parser, l7, labelsMap, deniedIdentities)
+			rule := getPortNetworkPolicyRule(sel, l4.L7Parser, l7, labelsMap, allowedIdentities, deniedIdentities)
 			if rule != nil {
 				if len(rule.RemotePolicies) == 0 && rule.L7Rules == nil {
```

Comment:
Is the L7Rules guaranteed to be nil or can it also be an empty slice?

Reply:
It is currently always nil when empty. But I agree that's brittle, so I'll change to a `len(rule.L7Rules) == 0` test.